tokio-current-thread/src/lib.rs

```diff
@@ -629,9 +630,9 @@ impl<'a, P: Park> fmt::Debug for Entered<'a, P> {
 /// Handle to spawn a future on the corresponding `CurrentThread` instance
 #[derive(Clone)]
 pub struct Handle {
-    sender: mpsc::Sender<Box<dyn Future<Item = (), Error = ()> + Send + 'static>>,
+    sender: crossbeam_channel::Sender<Box<dyn Future<Item = (), Error = ()> + Send + 'static>>,
     num_futures: Arc<atomic::AtomicUsize>,
-    shut_down: Cell<bool>,
+    shut_down: Arc<atomic::AtomicBool>,
```

Comment:
Why is `Arc` needed here? The `AtomicBool` could be stored directly in the struct?

Reply:
You're right, with the current behaviour (using `Cell<bool>`), when cloning a `Handle` a new boolean will be created, completely disconnected from the old one. With the new behaviour (using `Arc<AtomicBool>`) only the reference-counter will be cloned, but the underlying value will be the same.

I'm wondering now if the previous behaviour is really correct though. I can see that `shut_down` is used to "cache" the value from `num_futures`, and to check the status of the Handle/CurrentThread in `status()`. Since `num_futures` is behind an `Arc`, shouldn't `shut_down` also be? What am I missing?

Reply:
I believe `shut_down` is just a cache. You can *probably* get rid of it in favor of just "recomputing"?

Reply:
I think you're right, and it's probably even faster because on the "happy path" only 1 instead of 2 atomics are read.